flake.nix

```diff
@@ -72,6 +72,12 @@
             '';
 
             GEN_ARTIFACTS = "artifacts";
+            TYPST_VERSION =
+              let
+                rev = self.shortRev or "dirty";
```

Comment:
Wouldn't it perhaps be a better idea to keep the "bad case" as `"unknown hash"` for consistency?

Reply:
I believe that here `dirty` is meaningful (https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitglossary.html#def_dirty). It means that there are changes in the repo, but they are not committed, the binary out of these sources will reflect that using the word `dirty`.

Reply:
Yeah, I thought so, I just thought it'd be weird to distribute a binary with "dirty" as the version (at least an end user would find that weird). But, in hindsight, perhaps such binaries shouldn't be distributed anyway, so this could end up making more sense. I'll let @laurmaedje decide :wink: